tests/actions.go

```diff
@@ -833,39 +842,239 @@ func checkoutTag(tagName string) error {
+		"mode":         "restore",
+		"user":         "root",
+		"password":     to.Password,
+		"storage.size": "10Gi",
+	}
+	var buffer bytes.Buffer
+	for k, v := range sets {
+		set := fmt.Sprintf(" --set %s=%s", k, v)
+		_, err := buffer.WriteString(set)
+		if err != nil {
+			return err
+		}
+	}
+
+	setStr := buffer.String()
+	restoreName := fmt.Sprintf("%s-restore", from.ClusterName)
+	cmd := fmt.Sprintf("helm install -n %s --namespace %s /charts/%s/tidb-backup %s",
+		restoreName, to.Namespace, to.OperatorTag, setStr)
+	glog.Infof("install restore [%s]", cmd)
+	res, err := exec.Command("/bin/sh", "-c", cmd).CombinedOutput()
+	if err != nil {
+		return fmt.Errorf("failed to launch restore job: %v, %s", err, string(res))
+	}
+
 	return nil
 }
 
-func (oa *operatorActions) CheckAdHocBackup(info *TidbClusterInfo) error {
+func (oa *operatorActions) CheckRestore(from *TidbClusterInfo, to *TidbClusterInfo) error {
+	glog.Infof("begin to check restore backup cluster[%s] namespace[%s]", from.ClusterName, from.Namespace)
+	defer func() {
+		glog.Infof("check restore end cluster[%s] namespace[%s]", to.ClusterName, to.Namespace)
+	}()
+
+	jobName := fmt.Sprintf("%s-restore-test-backup", to.ClusterName)
+	fn := func() (bool, error) {
+		job, err := oa.kubeCli.BatchV1().Jobs(to.Namespace).Get(jobName, metav1.GetOptions{})
+		if err != nil {
+			glog.Errorf("failed to get jobs %s ,%v", jobName, err)
+			return false, nil
+		}
+		if job.Status.Succeeded == 0 {
+			glog.Error("cluster [%s] back up job is not completed, please wait! ", to.ClusterName)
+			return false, nil
+		}
+
+		fromCount, err := from.QueryCount()
+		if err != nil {
+			glog.Error("cluster [%s] count err ", from.ClusterName)
+			return false, nil
+		}
+
+		toCount, err := to.QueryCount()
+		if err != nil {
+			glog.Error("cluster [%s] count err ", to.ClusterName)
+			return false, nil
+		}
+
+		if fromCount != toCount {
+			glog.Error("cluster [%s] count %d cluster [%s] count %d is not equal ",
+				from.ClusterName, fromCount, to.ClusterName, toCount)
+			return false, nil
+		}
+		return true, nil
+	}
+
+	err := wait.Poll(DefaultPollInterval, DefaultPollTimeout, fn)
+	if err != nil {
+		return fmt.Errorf("failed to launch scheduler backup job: %v", err)
+	}
 	return nil
 }
 
-func (oa *operatorActions) DeployIncrementalBackup(from *TidbClusterInfo, to *TidbClusterInfo) error {
+func (oa *operatorActions) ForceDeploy(info *TidbClusterInfo) error {
+	if err := oa.CleanTidbCluster(info); err != nil {
+		return err
+	}
+
+	if err := oa.DeployTidbCluster(info); err != nil {
+		return err
+	}
+
 	return nil
 }
 
-func (oa *operatorActions) CheckIncrementalBackup(info *TidbClusterInfo) error {
+func (info *TidbClusterInfo) QueryCount() (int, error) {
+	tableName := "test"
+	db, err := sql.Open("mysql", getDSN(info.Namespace, info.ClusterName, "record", info.Password))
+	if err != nil {
+		return 0, err
+	}
+
+	rows, err := db.Query(fmt.Sprintf("SELECT count(*) FROM %s", tableName))
```

Comment:
```go
var count int
if err := db.QueryRaw(...).Scan(&count); err != nil {
...
}
return count, nil
```
otherwise, you have to close rows.

Reply:
It seems all right， I‘ll change it in next PR.